core/modules/syncer.js

```diff
@@ -129,47 +133,40 @@ function Syncer(options) {
 	});
 }
 
+/*
+Return an array of the tiddler titles that are subjected to syncing
+*/
+Syncer.prototype.getSyncedTiddlers = function(source) {
+	return this.filterFn.call(this.wiki,source);
+};
+
 /*
 Read (or re-read) the latest tiddler info from the store
 */
 Syncer.prototype.readTiddlerInfo = function() {
-	// Hashmap by title of {revision:,changeCount:,adaptorInfo:,needsToBeLoaded:,hasBeenLazyLoaded:}
+	// Hashmap by title of {revision:,changeCount:,adaptorInfo:}
+	// "revision" is the revision of the tiddler last seen on the server, and "changecount" is the corresponding local changecount
 	this.tiddlerInfo = {};
 	// Record information for known tiddlers
 	var self = this,
-		tiddlers = this.filterFn.call(this.wiki);
+		tiddlers = this.getSyncedTiddlers();
 	$tw.utils.each(tiddlers,function(title) {
 		var tiddler = self.wiki.getTiddler(title);
 		self.tiddlerInfo[title] = {
-			revision: tiddler.fields.revision,
+			revision: (self.syncadaptor && self.syncadaptor.getTiddlerRevision && self.syncadaptor.getTiddlerRevision(tiddler)) || tiddler.fields.revision,
```

Comment:
hmmm .. `self` is local to the function in line 218 self is "global"

Reply:
Ouch!